var/spack/repos/builtin/packages/python/package.py

```diff
@@ -151,6 +151,10 @@ class Python(AutotoolsPackage):
     # An in-source build with --enable-optimizations fails for python@3.X
     build_directory = 'spack-build'
 
+    def url_for_version(self, version):
+        url = "https://www.python.org/ftp/python/{0}/Python-{1}.tgz"
+        return url.format(re.split('[a-z]', str(version))[0], version)
```

Comment:
You should also be able to do:
```suggestion
        return url.format(version.up_to(3), version)
```

Reply:
Ah, but I see some versions like `3.2` with only 2 components. Your solution might be more robust in that case.